src/views/document/RouteView.vue

```diff
@@ -186,6 +188,25 @@ export default {
 
       return result;
     },
+    documentType() { // is-editable mixin needs this property
+        return this.$documentUtils.getDocumentType(this.document.type);
+      },
+    PromoteHistory() {
+        const result = {};
+        const doc = this.document;
+        const activities = doc.activities ?? [];
+        const history_worth_activities = ['snow_ice_mixed', 'mountain_climbing', 'rock_climbing', 'ice_climbing', 'via_ferrata', 'slacklining'];
```

Comment:
those consts could be declared outside of the vue export so it's not computed each time the function is called (I know it's a copy & paste and could be improved in other places too)